src/detect-ftpcommand.c

```diff
@@ -0,0 +1,278 @@
+/* Copyright (C) 2017 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Eric Leblond <eric@regit.org>
+ *
+ * Match on ftp command used to trigger a ftp data transfer
+ */
+
+#include "suricata-common.h"
+#include "util-unittest.h"
+
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-state.h"
+
+#include "app-layer-ftp.h"
+
+#include "detect-ftpcommand.h"
+
+/**
+ * \brief Regex for parsing our keyword options
+ */
+#define PARSE_REGEX  "^\\s*(stor|retr)\\s*$"
+static pcre *parse_regex;
+static pcre_extra *parse_regex_study;
+
+/* Prototypes of functions registered in DetectFtpcommandRegister below */
+static int DetectFtpcommandMatch(ThreadVars *, DetectEngineThreadCtx *,
+        Flow *, uint8_t, void *, void *,
+        const Signature *, const SigMatchCtx *);
+static int DetectFtpcommandSetup (DetectEngineCtx *, Signature *, const char *);
+static void DetectFtpcommandFree (void *);
+static void DetectFtpcommandRegisterTests (void);
+static int DetectEngineInspectFtpdataGeneric(ThreadVars *tv,
+        DetectEngineCtx *de_ctx, DetectEngineThreadCtx *det_ctx,
+        const Signature *s, const SigMatchData *smd,
+        Flow *f, uint8_t flags, void *alstate,
+        void *txv, uint64_t tx_id);
+static int g_ftpdata_buffer_id = 0;
+
+/**
+ * \brief Registration function for ftpcommand: keyword
+ *
+ * This function is called once in the 'lifetime' of the engine.
+ */
+void DetectFtpcommandRegister(void) {
+    /* keyword name: this is how the keyword is used in a rule */
+    sigmatch_table[DETECT_FTPCOMMAND].name = "ftpdata_command";
```

Comment:
as long as this does not allow matching on ftp commands in general, but only on the ftpdata_command, I would like to see the files, functions and keyword ID also reflect that in their names